gravitee-plugin-annotation-processors/README.adoc

```diff
@@ -0,0 +1,164 @@
+. Add this plugin in your pom.xml to defined the generated sources as source file (so you can use them in your classes in your IDE)
+
+[source, xml]
+----
+<plugin>
+    <groupId>org.codehaus.mojo</groupId>
+    <artifactId>build-helper-maven-plugin</artifactId>
+    <version>1.7</version>
+    <executions>
+        <execution>
+            <id>add-source</id>
+            <phase>generate-sources</phase>
+            <goals>
+                <goal>add-source</goal>
+            </goals>
+            <configuration>
+                <sources>
+                    <source>generated-sources</source>
+                </sources>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+----
+[start=4]
+. Add the annotation `@ConfigurationEvaluator` at the class level of your plugin configuration classes (ie: KafkaEndpointConnectorConfiguration, KafkaEndpointConnectorSharedConfiguration) with the right value for the parameter `attributePrefix`
+. Add the validation annotations in your configuration classes as described in <<_validate_the_dynamic_configuration>>
+. Build your project to generate the new "ConfigurationEvaluator" classes (you can find them in the folder target/generated-sources/annotations)
+. Create all `evaluators` necessary in your constructor (only one attribute is required by the evaluator constructor which is the configuration bean) and call the `eval` method to generate the dynamic configuration object or retrieved it from the context if it has already been called once (it is stored in the context using an internal attribute to avoid multiple evaluation). Here you have an example for an endpoint :
+
+[source, java]
+----
+    // parts of the code are missing to focus on what needs to be changed
+
+    @Override
+    public Completable subscribe(final ExecutionContext ctx) {
+        return kafkaEndpointConnectorConfigurationEvaluator
+            .eval(ctx)
+            .flatMapCompletable(evaluatedConfiguration ->
+                kafkaEndpointConnectorSharedConfigurationEvaluator
+                    .eval(ctx)
+                    .flatMapCompletable(evaluatedSharedConfiguration ->
+                        Completable.fromRunnable(() -> {
+                            if (evaluatedSharedConfiguration.getConsumer().isEnabled()) {
+                                ctx
+                                    .response()
+                                    .messages(
+                                            // ...
+                                    );
+                            }
+                        })
+                    )
+            );
+    }
+----
+[start=8]
+. Update your unit tests and you should be done
+
+=== Methods available in evaluator class
+
+The generated evaluator provides 3 methods that you can use :
+
+* public Completable eval() : default method to use which parse the original configuration and evaluate it in a reactive way then set the result as an internal attribute
```

Comment:
According to the example you gave above:

```java
return kafkaEndpointConnectorConfigurationEvaluator
            .eval(ctx)
            .flatMapCompletable(evaluatedConfiguration ->
                kafkaEndpointConnectorSharedConfigurationEvaluator
                    .eval(ctx)
                    .flatMapCompletable(evaluatedSharedConfiguration -> ...
```

I think there is something wrong because a completable does not return anything (use `andThen()`?).